File: 540. Single Element in a Sorted Array/540. Single Element in a Sorted Array.py
class Solution(object):
    def singleNonDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        i, lenn = 0, len(nums) - 1
        while i < lenn:
            if nums[i] != nums[i + 1]:
                return nums[i]
            else:
                i += 2
        return nums[-1]

File: 540. Single Element in a Sorted Array/540. Single Element in a Sorted Array.rb
# @param {Integer[]} nums
# @return {Integer}
def single_non_duplicate(nums)
    i = 0
    numsSize = nums.size - 1
    while i < numsSize
        if nums[i] != nums[i + 1]
            return nums[i]
        else
            i = i + 2
        end
    end
    return nums.last
end
